### PR TITLE
git-branchless: Add version 0.5.1

### LINKED ISSUE #3995: [Request] git-branchless
### Criteria

For a package to be acceptable in the Main bucket, it should be:

- [x] a non-GUI tool
- [x] a reasonably well-known and widely used developer tool (e.g. if it's a GitHub project, it should have at least 500 stars and 150 forks)
- [x] the latest stable version of the program
- [x] the full version i.e. not a trial version
- [x] a fairly standard install (e.g. uses a version-specific download URL, no elaborate pre/post install scripts)

### Information

**Name:** git-branchless

**Description:** High-velocity, monorepo-scale workflow for Git

**Homepage:** https://github.com/arxanas/git-branchless

**Download link(s):** https://github.com/arxanas/git-branchless/releases/download/v0.5.1/git-branchless-v0.5.1-x86_64-pc-windows-msvc.zip

**Some indication of popularity/repute:** Stars > 2.1k

### PR DESCRIPTION
**Pull request by Zliced13**
Closes #3995

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rashil2000** [comment]
This is a 64bit application

**Comment by Zliced13** [comment]
It's only available as a 64bit application. Why should I bother using the architecture block?

**Comment by rashil2000** [comment]
Read the Contributing guide:
> If the application provides only a 32bit download, the architecture field is not required. In all other cases, architecture field is mandatory.

**Comment by Zliced13** [comment]
Oh thats why ok.

File: bucket/git-branchless.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "architecture": {
+        "64bit": {
+            "url": "https://github.com/arxanas/git-branchless/releases/download/v0.5.1/git-branchless-v0.5.1-x86_64-pc-windows-msvc.zip",
+            "hash": "B7BCCE425374D8746364AE96B4DD6A4BD0917E28DDBD2665E1DA6E06BA10BF92"
+        }
+    },
+    "bin": "git-branchless.exe",
+    "checkver": "github",
+    "autoupdate": {
+        "url": "https://github.com/arxanas/git-branchless/releases/download/v$version/git-branchless-v$version-x86_64-pc-windows-msvc.zip"
```
Comment (rashil2000) [inline bucket/git-branchless.json:15]: Here too

Comment (Zliced13) [reply bucket/git-branchless.json]: fixed

**Pull request merged**